### ISSUE TITLE
Add a custom layout handle when there are no search results

### ISSUE THREAD

**Issue by jakwinkler** [magento/magento2#18038]
### Description

This is something frontend developers struggle with. When there are no search results, it would be nice to have a customized layout handle to add some custom blocks when there are no search results. 

### Expected behavior

- unique layout update when there are no search results found, it can be a blank file but it needs to be processed when layout is being loaded.
- ability to change page layout to 1,2 3 columns that does not affect search page with results

### Benefits

- customized handle to add content blocks / product blocks when products are not found (some elastic search modules already do that, but they are based on collection count)
- abiltiy to change layout of no-result page.
- less frustration when coding frontend search pages

### Additional information

I think the description is self-explanatory. 
Handle name can be `search_result_noresults` or something like that.

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@qsolutions-pl do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by progreg** [comment]
#lviv-meetup

**Comment by magento-engcom-team** [comment]
@progreg thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @progreg. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `G1 Passed` will be added to the issue automatically. Please, edit issue description if needed, until label `G1 Passed` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `G2 Passed` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `G3 Passed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report is acknowledged.

**Comment by jakwinkler** [comment]
I'm happy that PR was created to this but... it does not work as expected.
I've implemented this pull request to 2.2.5 version (EE), created a file **catalogsearch_result_noresults.xml** in Magento CatalogSearch module with following content
```
<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="1column"
      xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <referenceContainer name="header" remove="true" />
    <referenceContainer name="header-wrapper" remove="true" />
    <referenceContainer name="footer" remove="true" />
</page>
```

Am I doing something wrong? Basically the layout xml file is not processed at all... at least thats what I can tell.

For now I will simply create a redirect to a new controller where layout handles work fine.

**Comment by ishakhsuvarov** [comment]
@qsolutions-pl It's best to discuss PR in the appropriate ticket, increases chances that author will respond

**Comment by jakwinkler** [comment]
@ishakhsuvarov  got it :-)

**Comment by progreg** [comment]
@qsolutions-pl, please check PR :)

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
The issue has been fixed in magento/magento2#18069 by @progreg in 2.3-develop branch
Related commit(s):
 - [23666acfbe35a00fe5c5be2c1e0e95440f32310f](https://github.com/magento/magento2/commit/23666acfbe35a00fe5c5be2c1e0e95440f32310f)
 - [158a0fd7a2b452d05744312bdddde287b292e494](https://github.com/magento/magento2/commit/158a0fd7a2b452d05744312bdddde287b292e494)
 - [d2c2c0067d19b1624e757776695ad19be4c49f52](https://github.com/magento/magento2/commit/d2c2c0067d19b1624e757776695ad19be4c49f52)
 - [ac8d4262b653e3e532fe1960f74700c83381ce7a](https://github.com/magento/magento2/commit/ac8d4262b653e3e532fe1960f74700c83381ce7a)
 - [258703c7919df4a22fa3ae5885b1baf00b75971d](https://github.com/magento/magento2/commit/258703c7919df4a22fa3ae5885b1baf00b75971d)
 - [ca8b15bd7678c49ee4e1eecf9746a19a99d4a118](https://github.com/magento/magento2/commit/ca8b15bd7678c49ee4e1eecf9746a19a99d4a118)
 - [c1cbd005dd2980c82641afb4dd86484ae12397bf](https://github.com/magento/magento2/commit/c1cbd005dd2980c82641afb4dd86484ae12397bf)

The fix will be available with the upcoming 2.3.1 release.
